### ISSUE TITLE
New `max-positional-arguments` in new pylint version - failing CI-linter

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/simtools#1147]
The [nightly CI-Linter](https://github.com/gammasim/simtools/actions/runs/10977082564/job/30483432283) run failed due to a pylint update with a new rule on `max-positional-arguments` (see [here](https://pylint.readthedocs.io/en/stable/whatsnew/3/3.3/index.html#new-checks) and [here](https://github.com/pylint-dev/pylint/issues/9099)).

What do we do?

Simplest solution: move to newest version and add

```
# Maximum number of positional arguments for function / method (default=5.)
max-positional-arguments = 15
```

to the pypproject.toml file? This means we have to update everywhere pylint (otherwise you will get errors with an older version of pylint like `E0015: Unrecognized option found: max-positional-arguments (unrecognized-option)`).

Although I am not convinced that this rule should be so stringent, my suggestion would be to fix this and restrict the number of positional arguments to 5. In the long term it is easier when we follow default values.

**Comment by orelgueta** [comment]
If I understand this requirement correctly, there is no limit for named arguments, right?
That is, if we have 
```
func(
        arg1=arg1,
        arg2=arg2,
        arg3=arg3,
        arg4=arg4,
        arg5=arg5,
        arg6=arg6,
        arg7=arg7,
    )
```
That's allowed, right?

If so, I would vote for fixing all occurrences and just provide named arguments everywhere (which we generally try to do anyway).

**Comment by GernotMaier** [comment]
There is a limit on the maximum number of arguments, which we have increased a lot in pyproject.toml (I think 15 is excessive):

```
# Maximum number of arguments for function / method (default=5).
max-args = 15
```

But isn't the way we are using it the right way to indicate which arguments are required and which are optional?

**Comment by orelgueta** [comment]
But positional arguments are not necessarily related to required or optional, no?
You can provide required arguments out of order, as long as they are named.

I agree 15 is excessive, but 5 is too low. We can reduce the limit to 10 perhaps, but take this opportunity to also avoid using positional arguments in general (unless it is just 1-2 arguments).

**Comment by GernotMaier** [comment]
I think I don’t know how to indicate requirement arguments which do not appear as positional arguments. How would one do that?

**Comment by orelgueta** [comment]
Not at the definition side, only when calling the function. If at the definition side we have functions with more than 5 positional arguments, then I agree it is too much. When calling the function we can avoid it though and then I don't think it's a problem.

**Comment by GernotMaier** [comment]
The issue is that I don’t exactly understand the pylint complaint. E.g. in https://github.com/gammasim/simtools/blob/f9b4808b73a27561d7d98b3648f80dbf6a5bc33c/simtools/layout/array_layout.py#L55 It looks to me like 3 positional arguments and 6 others. Pylint are complaining about 9 positional arguments.

**Comment by orelgueta** [comment]
Yes, it probably doesn't like the fact we listed the arguments. But I wouldn't really call them positional ones if they have a default value and I much prefer that to `*args, **kargs`. I therefore vote to just increase the limit to 15.
